kv/db_test.go

```diff
@@ -130,14 +130,15 @@ func TestKVDBInternalMethods(t *testing.T) {
 		args  proto.Request
 		reply proto.Response
 	}{
-		{&proto.InternalRangeLookupRequest{}, &proto.InternalRangeLookupResponse{}},
-		{&proto.InternalGCRequest{}, &proto.InternalGCResponse{}},
-		{&proto.InternalHeartbeatTxnRequest{}, &proto.InternalHeartbeatTxnResponse{}},
-		{&proto.InternalPushTxnRequest{}, &proto.InternalPushTxnResponse{}},
-		{&proto.InternalResolveIntentRequest{}, &proto.InternalResolveIntentResponse{}},
-		{&proto.InternalResolveIntentRangeRequest{}, &proto.InternalResolveIntentRangeResponse{}},
-		{&proto.InternalMergeRequest{}, &proto.InternalMergeResponse{}},
-		{&proto.InternalTruncateLogRequest{}, &proto.InternalTruncateLogResponse{}},
+		{&proto.HeartbeatTxnRequest{}, &proto.HeartbeatTxnResponse{}},
```

Comment:
I added code to verify that batches don't contain non-public requests. As long as this HTTP endpoint is serving public API, might as well do it right. When we're really going to internal only and not exposing this endpoint, the KV API requests will go directly to the local dist sender, not through an HTTP endpoint.